### ISSUE TITLE
Use JDK only when Java source

### ISSUE THREAD

**Issue by squakez** [apache/camel-k#3743]
When running an Integration we always use a JDK based image. We always need a JDK/JRE based container as at the end of the day we run a Camel application. Probably we can simply detect if an application has some java source (in which case we use the JDK), otherwise we can use a JRE image.

_Originally posted by @squakez in https://github.com/apache/camel-k/issues/3424#issuecomment-1183210825_

**Comment by mdebarros** [comment]
Is a JDK based image also required when building "Quarkus native" Kits?

**Comment by squakez** [comment]
> Is a JDK base image also required when building "Quarkus native" Kits?

No, in that case it uses quarkus-micro-image. See https://camel.apache.org/camel-k/2.2.x/traits/quarkus.html to override the default if it is required.
